### ISSUE TITLE
Detect `git push --delete` tag action and fire webhook

### ISSUE THREAD

**Issue by LeKovr** [gogits/gogs#2315]
When tag created via `git tag v1 ; git push --tags` there is a hook called with such data:

```
{
 "ref": "v1",
 "ref_type": "tag",
 ...
}
```

But when you do `git push --delete origin v1` tag will be removed but no hook fired.

Version: 0.8.14.1230 (current on try.gogs.io)

**Comment by Unknwon** [comment]
Hi, I think Gogs does not detect delete tag event at all.

How does GitHub handle this?

**Comment by LeKovr** [comment]
> Gogs does not detect delete tag event at all.

Don't think so.
There is log for adding tag v4.21:
```
2016/01/01 12:58:01 .[1;36m[T] TriggerTask [1].(new request): refs/tags/v4.21.[0m
2016/01/01 12:58:01 .[1;36m[T] DeliverHooks [1]: processing delivery hooks.[0m
2016/01/01 12:58:01 .[1;36m[T] AddTestPullRequestTask[head_repo_id: 1, head_branch: refs/tags/v4.21]: finding pull requests.[0m
2016/01/01 12:58:01 .[1;36m[T] AddTestPullRequestTask[base_repo_id: 1, base_branch: refs/tags/v4.21]: finding pull requests.[0m
2016/01/01 12:58:02 .[1;36m[T] Hook delivered: 72ffd2ae-be8f-4b71-968f-41b62362fbe8.[0m
```

And there is log for deleting:
```
2016/01/01 12:59:09 .[1;36m[T] TriggerTask [1].(new request): refs/tags/v4.14.[0m
2016/01/01 12:59:09 .[1;36m[T] DeliverHooks [1]: processing delivery hooks.[0m
2016/01/01 12:59:09 .[1;36m[T] AddTestPullRequestTask[head_repo_id: 1, head_branch: refs/tags/v4.14]: finding pull requests.[0m
2016/01/01 12:59:09 .[1;36m[T] AddTestPullRequestTask[base_repo_id: 1, base_branch: refs/tags/v4.14]: finding pull requests.[0m
```

**Comment by Unknwon** [comment]
The log only shows there is something about `tag v4.2.1`, but it does not know it is delete or other actions.

Currently only add action is being detected.

**Comment by Unknwon** [comment]
I have turned this issue into a feature request, if that's OK to you.

**Comment by LeKovr** [comment]
Yes, it is ok. Thank you

**Comment by Unknwon** [comment]
Patch has pushed to fix this issue, please test on `release/0.11` branch or https://try.gogs.io.
